src/app/content/hooks/receiveContent.ts

```diff
@@ -50,19 +51,22 @@ const hookBody: ActionHookBody<typeof receivePage> = ({
   const canonicalUrl = canonical && contentRoute.getUrl(canonical);
   const bookTheme = theme.color.primary[hasOSWebData(book) ? book.theme : defaultTheme].base;
 
-  dispatch(setHead({
-    links: canonicalUrl ? [
-      {rel: 'canonical', href: `https://openstax.org${canonicalUrl}`},
-    ] : [],
-    meta: [
-      {name: 'description', content: description},
-      {property: 'og:description', content: description},
-      {property: 'og:title', content: title},
-      {property: 'og:url', content: `https://openstax.org${currentPath}`},
-      {name: 'theme-color', content: bookTheme},
-    ],
-    title,
-  }));
+  const links = canonicalUrl ? [
+    {rel: 'canonical', href: `https://openstax.org${canonicalUrl}`} as Link,
+  ] : [];
+  const meta = [
+    {name: 'description', content: description},
+    {property: 'og:description', content: description},
+    {property: 'og:title', content: title},
+    {property: 'og:url', content: `https://openstax.org${currentPath}`},
+    {name: 'theme-color', content: bookTheme},
+  ];
+
+  if (hasOSWebData(book) && book.promote_image) {
```

Comment:
Just curious, why is this `hasOSWebData` check necessary?

Reply:
because `book` is of type `Book` which is `BookWithOSWebData | ArchiveBook` and without this guard typescript will not allow us to access `promote_image` property that is only available on `BookWithOSWebData`